plugin/src/main/kotlin/io/github/gmazzo/android/test/aggregation/AndroidTestCoverageAggregationPlugin.kt

```diff
@@ -105,21 +113,20 @@ abstract class AndroidTestCoverageAggregationPlugin : Plugin<Project> {
                     objects.named(VerificationType.JACOCO_RESULTS)
                 )
             }
-            jacocoVariants.all variant@{
-                val execData = tasks
-                    .named("test${this@variant.unitTest!!.name.capitalized()}")
-                    .map { it.the<JacocoTaskExtension>().destinationFile!! }
-
-                outgoing.artifact(execData) {
-                    type = ArtifactTypeDefinition.BINARY_DATA_TYPE
+            codeCoverageExecData.all file@{
+                files(this).asFileTree.forEach { file ->
+                    outgoing.artifact(file) {
+                        type = ArtifactTypeDefinition.BINARY_DATA_TYPE
+                        builtBy(this@file)
+                    }
                 }
```

Comment:
And this is not working well, as for `AndroidTest` we know in advance the target output directory, but not all the `.ec` files getting generated by each emulator.